utils/crypto/bls/secret.go

```diff
@@ -19,55 +19,69 @@ var (
 	// The ciphersuite is more commonly known as G2ProofOfPossession.
 	// There are two digests to ensure that message space for normal
 	// signatures and the proof of possession are distinct.
-	ciphersuiteSignature         = []byte("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
-	ciphersuiteProofOfPossession = []byte("BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+	ciphersuiteSignature                = []byte("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+	ciphersuiteProofOfPossession        = []byte("BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+	_                            Signer = (*LocalSigner)(nil)
 )
 
 type SecretKey = blst.SecretKey
 
+type Signer interface {
+	PublicKey() *PublicKey
+	Sign(msg []byte) *Signature
+	SignProofOfPossession(msg []byte) *Signature
```

Comment:
I feel like these functions are going to need to return `error` once we want to use them for remote signing. Are we waiting to add `error` to the return value for when that other implementation is introduced?

Reply:
Yeah, I don't think there's any point in pre-maturely defining the interface here. It's unclear to me at this point whether or not we'll want the specific signer implementation to handle its own errors or wether we would want the caller to handle errors (I'm missing some details on how to gracefully shutdown). 

I'm interested to hear your thoughts here, but I definitely don't think this is something that we should deal with yet.

Reply:
Sounds good. I think the only complex question will be around the PoP in the networking code... Everything else should be fairly straightforward I think... But 🤷 we can deal with that when we need to I suppose.